### PR TITLE
New mention plugin

### PR DESCRIPTION
**Pull request by Globulopolis**

### CODE REVIEW

**Comment by Globulopolis** [comment]
Maybe will close https://github.com/Alex-D/Trumbowyg/issues/537

How to use.
Insert plugin script

```html
<script src="trumbowyg.mention.js"></script>
```

Init editor and add button in config.

```javascript
var src = [
	{
		'login': 'Foo',
		'url': 'http://google.com'
	},
	{
		'login': 'Bar',
		'url': 'http://google.com'
	},
	{
		'login': 'Baz',
		'url': 'http://google.com'
	}
];
var ed = $('#default-editor').trumbowyg({
	lang: 'ru',
	btns: [
		['formatting'],
		['viewHTML'],
		['mention']
	],
	svgPath: 'dist/ui/icons.svg',
	plugins: {
		'mention': {
			source: responce,
			formatDropdownItem: function(item){
				return '<strong>' + item.login + '</strong> ' + item.login;
			},
			formatResult: function(item){
				return '@<a href="' + item.url + '">' + item.login + '</a>';
			}
		}
	}
});
```

where:
`source` - array with objects. For more info about structure see https://developer.github.com/v3/users/#get-all-users
`formatDropdownItem` - callback to format item in dropdown list. Default `item.login`
`formatResult` - callback to format result in editor. Default `'@' + item.login + ' '`

**NB!**
If you want to fill source from ajax request - init editor directly in success(done()) callback.

File: plugins/mention/trumbowyg.mention.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+     * @param   {Array}   items      Items
+     * @param   {object}  trumbowyg  Editor
+     *
+     * @return  {Array}
+     */
+    function buildDropdown(items, trumbowyg) {
+        var dropdown = [];
+
+        // Check if source is an array
+        if (trumbowyg.o.plugins.mention.source.constructor === Array) {
```
Comment (Alex-D) [inline plugins/mention/trumbowyg.mention.js]: Should use `items`

Comment (Globulopolis) [reply plugins/mention/trumbowyg.mention.js]: What means `items`?

Comment (Alex-D) [reply plugins/mention/trumbowyg.mention.js]: Your function argument :)

Comment (Alex-D) [reply plugins/mention/trumbowyg.mention.js]: `dropdown: buildDropdown(trumbowyg.o.plugins.mention.source, trumbowyg)`

`function buildDropdown(items, trumbowyg) {`

So here you should write:

`if (items.constructor === Array) {`

**Comment by Globulopolis** [comment]
Ah :) I'll fix it tomorrow.

**Comment by Novocaine85** [comment]
Sorry to being such a pain in the ass, but you really need to put an example on the documentation about this.

I spent all day to figure out how to make it work with ajax...

`var users  = $.getJSON({url: "yoururl.whatever"}).done(function(res) {
        $('.yourclass').trumbowyg({
            btns: [
                ['mention'],
            ],
            plugins: {
                'mention': {
                    source: res,
                    formatDropdownItem: function(item){
                        return '<strong>' + item.login + '</strong>';
                    },
                    formatResult: function(item){
                        return '@'+item.login+' ';
                    }
                }
            }
        })
    });`

Hope this code above will clarify if someone should need this plugin.

Thank you and sorry again <3

**Pull request merged**